cg/meta/archive/ddn_dataflow.py

```diff
@@ -41,11 +46,22 @@ class MiriaFile(FileTransferData):
     @classmethod
     def from_file_and_sample(
         cls, file: File, sample: Sample, is_archiving: bool = True
-    ) -> "MiriaFile":
+    ) -> "MiriaObject":
         """Instantiates the class from a File and Sample object."""
         if is_archiving:
-            return cls(destination=sample.internal_id, source=file.path)
-        return cls(destination=file.path, source=sample.internal_id)
+            return cls(destination=sample.internal_id, source=file.full_path)
+        return cls(destination=file.full_path, source=sample.internal_id)
+
+    @classmethod
+    def from_sample_and_destination(
```

Comment:
Feel free to disregard
```suggestion
    def create_from_sample_and_destination(
```